src/requirements/Payment/PaymentForm.tsx

```diff
@@ -58,19 +58,14 @@ const PaymentForm = ({
     token,
     fee,
     owner,
-    onSuccess: (registeredVaultId) =>
-      setValue(`${baseFieldPath}.data.id`, registeredVaultId),
+    onSuccess: (registeredVaultId) => {
+      setValue(`${baseFieldPath}.data.id`, registeredVaultId)
+      if (registeredVaultId) {
```

Comment:
I believe `registeredVaultId` will always be defined here (see the `useRegisterVault` hook for more details), so it's safe to remove this check.